rust/connlib/tunnel/src/client.rs

```diff
@@ -223,9 +193,73 @@ pub struct ClientState {
     active_candidate_receivers: StreamMap<GatewayId, RTCIceCandidateInit>,
     /// We split the receivers of ICE candidates into two phases because we only want to start sending them once we've received an SDP from the gateway.
     waiting_for_sdp_from_gatway: HashMap<GatewayId, Receiver<RTCIceCandidateInit>>,
+
+    // TODO: Make private
```

Comment:
Open TODO.

Reply:
@conectado So I tried to make these private but I am worried that this PR will get massive if I continue on. The issue is that the `on_open` handler for data channels also needs to access this which is just a bigger refactoring.

I think this PR is valuable as is so I'd like to leave this for a follow-up!